internal/config/config.go

```diff
@@ -4,12 +4,13 @@ import (
 	"strings"
 	"time"
 
+	"github.com/spf13/pflag"
 	"github.com/spf13/viper"
 )
 
 const (
-	FileName = "nri-kubernetes-config"
-	FilePath = "/etc/newrelic-infra/integrations.d/"
```

Comment:
I read the issue but I did not understand why the location is invalid and what you mean by "That path is for the agent, not the integration itself."
The charts looks like it is mounting the file exacting there and so far always worked.

Reply:
I made a PR for the helm chart as well: https://github.com/newrelic/helm-charts/pull/638

`/etc/newrelic-infra/integrations.d/` is the path were the agent read the configurations for integrations, not the config for the integration itself.

As the agent is if forward only mode there is so much issue with that but in the case of kubelet daemonset, the agent compalins of that  file not being a valid config v3/v4 config file.

TL;DR: Agent expect [this](https://docs.newrelic.com/docs/infrastructure/host-integrations/infrastructure-integrations-sdk/specifications/host-integrations-standard-configuration-format/#config) instead of the configuration for the integration itself.

Reply:
The chart should not mount that as well in the agent sidecar, therefore  the agent should not aware of this at all.

However, it is true what you say and `/etc/newrelic-infra` looks better

Reply:
I am rookie so I could be wrong. Correct me if I am wrong, please.

The agent does not need of this file because is a configuration for the integration itself, the agent should not have any configuration file for this integration as it should not run/execute any binary for this integration.

What the agent should have is an open port were this integration will POST the metrics that the agent should send to the backend.

AFAIK the agent do not need anything for that except forwardMode and forwardOnlyMode, right?

Reply:
everything correct 😄 
The agent container should not be even aware of this file, therefore you were right that `integrations.d/` looks weird